### PR TITLE
Resubmit of #2730

### PR DESCRIPTION
**Pull request by jjatie**
Resubmit of #2730

### CODE REVIEW

**Comment by petester42** [comment]
With the revert to isEqual seems like methods are just moved around. Is there any meaningful change anymore?

**Comment by jjatie** [comment]
- Use of `==` in the framework as per Swift guidelines
- removed the `==` implementations as noted by the Objective-C/Swift interoperability guide
- simplified the `isEqual` implementations for `ChartDataEntryBase` and `Highlight`
- removed dependence of the super class in the `ChartDataEntry` `isEqual` implementation

**Comment by jjatie** [comment]
@petester42 Any further feedback on this one or can it be merged?

**Comment by liuxuan30** [comment]
@jjatie we still need a review process for what coming into master.

**Comment by jjatie** [comment]
Agreed. This was a PR that has had feedback back and forth since August. On the PR this one replaced, we agreed this would be merged in after the Swift 4 migration. I made the adjustments requested from the review, and this has just been sitting here since.

This PR removes a few mistaken `==` implementations, and takes advantage of the Swift `Equatable` protocol without changing any implementations. Seemed safe to merge after all this time.

File: Tests/Charts/EquatableTests.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
+//
+//  EquatableTests.swift
```
Comment (liuxuan30) [inline Tests/Charts/EquatableTests.swift]: I made the same template as yours for combined chart test. Shall we include the license info just like source code template, or test files is fine with default Xcode template?

Comment (jjatie) [reply Tests/Charts/EquatableTests.swift]: I guess we should be putting in the license info to be consistent. Do we keep it the same, but whose name do we use?

Comment (liuxuan30) [reply Tests/Charts/EquatableTests.swift]: I guess just the same template as source code. I don't know the rule of names. 
Personally I just use `Copyright 2015 Daniel Cohen Gindi & Philipp Jahoda`, and I don't need to specify who created the file. 
I think this is a good question, as I saw some files with yours template:
```
//
//  DemoBaseViewController.swift
//  ChartsDemo-iOS
//
//  Created by Jacob Christie on 2017-07-03.
//  Copyright © 2017 jc. All rights reserved.
//
```
I guess we need an unified one. @petester42 @danielgindi @PhilJay insights?

Comment (jjatie) [reply Tests/Charts/EquatableTests.swift]: Yea, I just use the Xcode default.

**Comment by danielgindi** [comment]
Just one thing: Please conform to the overall style of the library.
I created this library with braces in new lines as I believe it makes for the cleanest and most readable code, and when there's mixed use of braces up and down - it creates a braces hell.

In any codebase that I participate - I use the style of that codebase, braces up or down, spaces or no spaces, var naming etc.

**Comment by liuxuan30** [comment]
@danielgindi how about the file template about license and names?

**Pull request merged**